### PR TITLE
Add Django 5.0 support, bump to django-nyt 1.4

### LINKED ISSUE #1330: Django 5 support
Pending other dependencies

* https://github.com/django-wiki/django-nyt/issues/126

### PR DESCRIPTION
**Pull request by benjaoming**
Fixes #1330 

Let's see how this goes in CI...

### CODE REVIEW

**Comment by benjaoming** [comment]
@oscarmcm ready for your eyes in case you have time :heart:

**Comment by kokospapa8** [comment]
Hi, I am wondering when will this feature be released? Is it going to be near future? 
I would like to use this library and wondering if I should just use django 4.x instead of 5.0.

**Comment by benjaoming** [comment]
Yes, sorry that this is dragging along. Will get back to it soon. It's something I've been wanting to do for a while.

**Comment by benjaoming** [comment]
@oscarmcm this is ready to be reviewed :+1:

File: .circleci/config.yml [hunk 1]
```diff
@@ -34,36 +34,58 @@ jobs:
...
+      - hatch:
+          name: "Python 3.11, Django 4.1"
           hatch_env: "py3.11-dj4.1"
           python_version: "3.11"
       - hatch:
+          name: "Python 3.11, Django 4.2"
           hatch_env: "py3.11-dj4.2"
           python_version: "3.11"
+      - hatch:
+          name: "Python 3.11, Django 5.0"
```
Comment (oscarmcm) [inline .circleci/config.yml:84]: niceeee

**Review by oscarmcm — APPROVED** [review]

**Pull request merged**